src/components/sections/CircuitSelect/index.tsx

```diff
@@ -46,10 +46,19 @@ const CircuitSelect = ({ appState, setAppState, repeats, setRepeats }: CircuitSe
         const data = await readFile(file)
         if (file.name.slice(-5) == ".json") {
             const json_data = JSON.parse(data as string)
-            setAppState({
-                apiResponse: new CompilationResultSuccess(json_data, ""),
-                compilationIsLoading: false,
-            })
+            if ((data as string).includes("compilation_text")) {
```

Comment:
I think here you might want to check that `json_data` has the `compilation_text` key, instead of checking the string ... otherwise you'll get strange bugs if `compilation_text` ends up in the qubit state (not sure why that would happen :joy: )

Reply:
Good point!